tests/e2e/test_syntax.py

```diff
@@ -75,6 +75,50 @@ def f(m1: MemRef[SInt16, 5]) -> Tuple[SInt16, SInt16]:
     assert f(np.zeros((5,), dtype=np.int16)) == (2 + 2 + 2, 8 + 8 + 8)
 
 
+def test_minus_eq():
+    @compile()
+    def f(a: SInt16) -> SInt16:
+        a -= 2
+        return a
+
+    assert f(1) == 1 - 2
+
+
+def test_plus_eq_memref():
+    @compile()
+    class Mod:
+        def f(m: MemRef[UInt8, 2, 3]) -> None:
+            m[1, 2] += 3
+
+    m = np.zeros((2, 3), dtype=np.uint8)
+
+    Mod.f(m)
+    expected = np.zeros_like(m)
+    expected[1, 2] = 3
+    assert np.array_equal(m, expected)
```

Comment:
```suggestion
    assert (m == expected).all()
```
Since you updated the other one as well.